src/database/queries.ts

```diff
@@ -105,11 +150,50 @@ export const getVoucher = async (uuid: Uuid): Promise<Voucher> => {
   }
 };
 
-export const createVoucher = async (voucher: VoucherCreate): Promise<Uuid> => {
+/**
+ * Query to create a new voucher in Firebase.
+ *
+ * Parameters: a json with fields
+ *
+ *    `serialNumber`: voucher serial number
+ *
+ *    `value`: monetary value of voucher in cents
+ *
+ *    `vendorUuid`: current user's Uuid
+ *
+ * Returns the doc id if the query executes successfully.
+ *
+ * Returns `'-1'` if the serial number is invalid,
+ * `'-2'` if the serial number has already been used,
+ * or `'-3'` if the voucher's value exceeds the maximum possible value.
+ */
```

Comment:
This also makes the error handling more intuitive, rather than having
```
const uuid = createVoucher ...`
```
where `uuid` could potentially be an error code

Reply:
Some docs the error handling method I proposed:
https://imhoff.blog/posts/using-results-in-typescript
https://journal.plain.com/posts/2022-10-04-error-handling-in-typescript-like-a-pro/
https://blog.logrocket.com/improve-error-handling-typescript-exhaustive-type-checking/